External/FEXCore/Source/Interface/Core/ArchHelpers/CodeEmitter/Registers.h

```diff
@@ -15,6 +18,8 @@ namespace FEXCore::ARMEmitter {
       constexpr explicit Register(uint32_t Idx)
         : Index {Idx} {}
 
+      friend constexpr auto operator<=>(const Register&, const Register&) = default;
```

Comment:
What use does `friend` have in this case?

Reply:
In the event we add comparisons with other register types (for whatever reason), then it stays consistent, since

```cpp
friend bool operator==(OtherRegisterType, RegisterType) { /* ... */ }
```
and
```cpp
friend bool operator==(RegisterType, OtherRegisterType) { /* ... */ }
```

would need to be handled. There's also nothing necessitating the comparison be bound to the class, so friend it was.

Reply:
> In the event we add comparisons with other register types (for whatever reason), then it stays consistent

Fair.

> There's also nothing necessitating the comparison be bound to the class, so friend it was.

Does `friend` have a difference as used here versus not using `friend`? This sentence suggests it does, but I don't see it.

Reply:
what I meant was, that coupled with my first point, I didn't really see any necessitating reason to tie the function to the class as a member function as opposed to a free function

Reply:
Ah, right. I guess I'm mostly used to seeing such operators defined out-of-class rather than in-class, but thanks for clearing it up!

Reply:
Oh, no prob!